t/t3431-rebase-ignore-ws.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/sh
+#
+# Copyright (c) 2019 Rohit Ashiwal
+#
+
+test_description='git rebase --ignore-whitespace tests'
+
+. ./test-lib.sh
+
+test_expect_success 'setup' '
+	git checkout -b topic &&
+	q_to_tab >main.c <<-EOF &&
+	int main() {
+	Qprintf("Hello World\n");
+	Qreturn 0;
+	}
+	EOF
+	git add main.c &&
+	git commit -m "add main.c" &&
+	q_to_tab >main.c <<-EOF &&
+	int main() {
+	Qprintf("Hello Git\n");
+	Qreturn 0;
+	}
+	EOF
+	git commit -am "update main.c" &&
+	git tag side &&
+
+	git checkout --orphan master &&
+	cat >main.c <<-EOF &&
+	int main() {
+	        printf("Hello World\n");
+	        return 0;
+	}
+	EOF
+	git add main.c &&
+	git commit -m "add main.c" &&
+	git tag main
+'
+
+test_expect_success '--ignore-whitespace works with am backend' '
+	test_must_fail git rebase main side &&
+	git rebase --abort &&
+	git rebase --ignore-whitespace main side
```

Comment:
Maybe check that this had the right result?

Reply:
:thinking: we can, but `--ignore-whitespace` in both cases will produce slightly different results. (but no error ofc)

Reply:
Ah right, and that's something we want to fix eventually, so it's better not to encode it in a test. Just testing for success makes sense to me then